lib/bugsnag/breadcrumbs/validator.rb

```diff
@@ -0,0 +1,47 @@
+module Bugsnag::Breadcrumbs
+  class Validator
+    ##
+    # Create a Validator object with the current configuration setup.
+    def initialize(configuration)
+      @configuration = configuration
+    end
+
+    ##
+    # Validates a given breadcrumb.
+    def validate(breadcrumb)
+      # Check message length
+      if breadcrumb.message.size > MAX_MESSAGE_LENGTH
+        @configuration.warn("Breadcrumb message trimmed to length #{MAX_MESSAGE_LENGTH}.  Original message: #{breadcrumb.message}")
+        breadcrumb.message.slice!(MAX_MESSAGE_LENGTH...breadcrumb.message.size)
```

Comment:
If it's passed to the breadcrumb initializer it should be a separate value, correct?  So even if it's mutated here, it won't effect anything upstream.

Reply:
Strings in ruby are mutable and passedin by reference. So as you currently have it it will mutate the argument passed in.

```ruby
a = "hello there this string is over 30 characters long as an example"
Bugsnag.leave_breadcrumb(a)
puts a
```

Will print "hello there this string is ove"